src/ballot.rs

```diff
@@ -43,3 +43,21 @@ impl Ballot {
     }
 }
 
+// normalize the max rating to 1, and everything else scaled proportionally
+pub fn normalize(student: Student) -> Student {
+    // find max
+    let mut max: f64 = 0.0;
+    for i in 0..student.ballot.len(){
```

Comment:
You can iterate over vectors in rust like this (rather than iterate over the index of the arrays):
```rust
for elm in array {
}
```

Reply:
I tried to put that in, but it then transferred the location of student.ballot, which meant that I couldn't iterate through later in the code. To avoid that I would have to make a duplicate of student.ballot, which would end up just wasting memory.

Reply:
Ah, Rust pointers are good for this:

Example:
```rust
fn main() {
    let array = vec![1, 2, 3];
    
    for num in &array {
        println!("{}", num);
    }
    for num in &array {
        println!("{}", num);
    }
}
```

https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=e11fb2e65a060f13b097a5318cf8834b